neard/tests/rpc_nodes.rs

```diff
@@ -446,3 +454,143 @@ fn test_get_validator_info_rpc() {
         system.run().unwrap();
     });
 }
+
+fn outcome_view_to_hashes(outcome: &ExecutionOutcomeView) -> Vec<CryptoHash> {
+    let status = match &outcome.status {
+        ExecutionStatusView::Unknown => PartialExecutionStatus::Unknown,
+        ExecutionStatusView::SuccessValue(s) => {
+            PartialExecutionStatus::SuccessValue(from_base64(s).unwrap())
+        }
+        ExecutionStatusView::Failure(_) => PartialExecutionStatus::Failure,
+        ExecutionStatusView::SuccessReceiptId(id) => PartialExecutionStatus::SuccessReceiptId(*id),
+    };
+    let mut result = vec![hash(
+        &(outcome.receipt_ids.clone(), outcome.gas_burnt, status)
+            .try_to_vec()
+            .expect("Failed to serialize"),
+    )];
+    for log in outcome.logs.iter() {
+        result.push(hash(log.as_bytes()));
+    }
+    result
+}
+
+fn test_get_execution_outcome(is_tx_successful: bool) {
+    init_integration_logger();
+    heavy_test(|| {
+        let system = System::new("NEAR");
+        let num_nodes = 2;
+        let dirs = (0..num_nodes)
+            .map(|i| {
+                tempfile::Builder::new().prefix(&format!("tx_propagation{}", i)).tempdir().unwrap()
+            })
+            .collect::<Vec<_>>();
+        let (genesis_config, rpc_addrs, clients) = start_nodes(1, &dirs, 1, 1, 1000, 0);
+        let view_client = clients[0].1.clone();
+
+        let genesis_hash = genesis_block(genesis_config).hash();
+        let signer = InMemorySigner::from_seed("near.0", KeyType::ED25519, "near.0");
+        let transaction = if is_tx_successful {
+            SignedTransaction::send_money(
+                1,
+                "near.0".to_string(),
+                "near.1".to_string(),
+                &signer,
+                10000,
+                genesis_hash,
+            )
+        } else {
+            SignedTransaction::create_account(
+                1,
+                "near.0".to_string(),
+                "near.1".to_string(),
+                10,
+                signer.public_key.clone(),
+                &signer,
+                genesis_hash,
+            )
+        };
+
+        WaitOrTimeout::new(
+            Box::new(move |_ctx| {
+                let client = new_client(&format!("http://{}", rpc_addrs[0]));
+                let bytes = transaction.try_to_vec().unwrap();
+                let view_client1 = view_client.clone();
+                actix::spawn(client.broadcast_tx_commit(to_base64(&bytes)).then(move |res| {
```

Comment:
Is there any issue with using async-await? The code will be much more straightforward

Reply:
Not sure. It's how other tests are written so I just followed their pattern.